java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -357,379 +349,87 @@ public static void jdbcToArrowVectors(ResultSet rs, VectorSchemaRoot root, Calen
   public static void jdbcToArrowVectors(ResultSet rs, VectorSchemaRoot root, JdbcToArrowConfig config)
       throws SQLException, IOException {
 
-    Preconditions.checkNotNull(rs, "JDBC ResultSet object can't be null");
-    Preconditions.checkNotNull(root, "VectorSchemaRoot object can't be null");
-    Preconditions.checkNotNull(config, "JDBC-to-Arrow configuration cannot be null");
-
     ResultSetMetaData rsmd = rs.getMetaData();
     int columnCount = rsmd.getColumnCount();
-
     allocateVectors(root, DEFAULT_BUFFER_SIZE);
 
+    JdbcConsumer[] consumers = new JdbcConsumer[columnCount];
```

Comment:
Given the potential resource leak below, I think it might make sense to make JdbcConsumer extend AutoCloseable and create a CompositeJdbcConsumer, that encompasses this logic and the closing below.

Reply:
Thanks for your reasonable suggestions, fixed now. Same should be done with avro adapter later.